src/server/trace/recorder/tracing.ts

```diff
@@ -230,4 +225,70 @@ export class Tracing implements InstrumentationListener {
       await fs.promises.appendFile(this._traceFile!, JSON.stringify(event) + '\n');
     });
   }
+
+  private _shouldCaptureSnapshot(metadata: CallMetadata): boolean {
+    return protocolMethodsToCapture.has(metadata.type + '.' + metadata.method);
+  }
 }
+
+const protocolMethodsToCapture = new Set([
+  'BrowserContext.newPage',
+  'ElementHandle.evalOnSelector',
+  'ElementHandle.evalOnSelectorAll',
+  'ElementHandle.check',
+  'ElementHandle.click',
+  'ElementHandle.dblclick',
+  'ElementHandle.dispatchEvent',
+  'ElementHandle.fill',
+  'ElementHandle.focus',
+  'ElementHandle.hover',
+  'ElementHandle.press',
+  'ElementHandle.scrollIntoViewIfNeeded',
+  'ElementHandle.selectOption',
+  'ElementHandle.selectText',
+  'ElementHandle.setInputFiles',
+  'ElementHandle.tap',
+  'ElementHandle.type',
+  'ElementHandle.uncheck',
+  'ElementHandle.waitForElementState',
+  'ElementHandle.waitForSelector',
+  'Frame.evalOnSelector',
+  'Frame.evalOnSelectorAll',
+  'Frame.addScriptTag',
+  'Frame.addStyleTag',
+  'Frame.check',
+  'Frame.click',
+  'Frame.dblclick',
+  'Frame.dispatchEvent',
+  'Frame.evaluateExpression',
+  'Frame.evaluateExpressionHandle',
+  'Frame.fill',
+  'Frame.focus',
+  'Frame.goto',
+  'Frame.hover',
+  'Frame.press',
+  'Frame.selectOption',
+  'Frame.setContent',
+  'Frame.setInputFiles',
+  'Frame.tap',
+  'Frame.type',
+  'Frame.uncheck',
+  'Frame.waitForFunction',
+  'Frame.waitForSelector',
+  'JSHandle.evaluateExpression',
+  'JSHandle.evaluateExpressionHandle',
+  'Page.goBack',
+  'Page.goForward',
+  'Page.reload',
+  'Page.setViewportSize',
+  'Page.keyboardDown',
+  'Page.keyboardUp',
+  'Page.keyboardInsertText',
+  'Page.keyboardType',
+  'Page.keyboardPress',
+  'Page.mouseMove',
+  'Page.mouseDown',
+  'Page.mouseUp',
+  'Page.mouseClick',
+  'Page.touchscreenTap',
```

Comment:
I'm not a huge fan of this list, we keep adding new APIs and then we might forget one to add there. :/

Reply:
I am with you, but I have not figured out anything better.

Reply:
Would it be better to have a blacklist that filters out instead of a whitelist that filters in?  Maybe even allow that list to be overridden by playwright.config?